pkg/build/strategies/sti/sti.go

```diff
@@ -496,6 +533,7 @@ func (builder *STI) Save(config *api.Config) (err error) {
 	extractFunc := func(string) error {
 		extractErr := builder.tar.ExtractTarStream(artifactTmpDir, outReader)
 		io.Copy(ioutil.Discard, outReader) // must ensure reader from container is drained
+
```

Comment:
this is a step that i would expect to have its own name and timing (RetrievePreviousArtifacts or something)